### PR TITLE
Remove Cache v23 support

### LINKED ISSUE #5092: Remove cache v23 support
It seems we have been supporting cache v23 for a really long time since ATS2.0. And I don't believe that there is someone still use such old version.  So do we still need to support this version ?

### PR DESCRIPTION
**Pull request by scw00**
This closes #5092

### CODE REVIEW

**Comment by scw00** [comment]
[approve ci freebsd]

**Comment by randall** [comment]
[approve ci freebsd]

**Comment by zwoop** [comment]
There are merge conflicts on this PR.

File: iocore/cache/I_CacheDefs.h [hunk 1]
```diff
@@ -114,8 +114,7 @@ enum CacheDataType {
 
 enum CacheFragType {
   CACHE_FRAG_TYPE_NONE,
-  CACHE_FRAG_TYPE_HTTP_V23, ///< DB version 23 or prior.
-  CACHE_FRAG_TYPE_RTSP,     ///< Should be removed once Cache Toolkit is implemented.
+  CACHE_FRAG_TYPE_RTSP, ///< Should be removed once Cache Toolkit is implemented.
   CACHE_FRAG_TYPE_HTTP,
```
Comment (SolidWallOfCode) [inline iocore/cache/I_CacheDefs.h]: Doesn't this change the value of `CACHE_FRAG_TYPE_HTTP`? That would seem to be a problem if the disk image has a fragment type of `4` (the current value) but the code expects a value of `2`.

Comment (scw00) [reply iocore/cache/I_CacheDefs.h]: You are right. That's would be a horrible.

Comment (scw00) [reply iocore/cache/I_CacheDefs.h]: So let's keep this CACHE_FRAG_TYPE_HTTP_V23 for not.

**Comment by zwoop** [comment]
This seems fine to me, you ok with this now @SolidWallOfCode ?

File: iocore/cache/I_CacheDefs.h (hunk 1 above)
Comment (zwoop) [reply iocore/cache/I_CacheDefs.h]: Another option is of course to remove the old enum symbols, and code the values for the remaining ones accordingly. I’m ok either way.

Comment (SolidWallOfCode) [reply iocore/cache/I_CacheDefs.h]: I would keep the enum and use `== 4`. It's possible we'll have other types in the future.

Comment (SolidWallOfCode) [reply iocore/cache/I_CacheDefs.h]: You might put in `CACHE_FRAG_TYPE_UNUSED_1`, etc., to occupy the space as well, just to be very clear.

**Comment by scw00** [comment]
I'm ok with both of them.  Updated

**Comment by scw00** [comment]
[approve ci]

**Comment by bryancall** [comment]
[approve ci]

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**